channelserver/src/server.rs

```diff
@@ -26,6 +27,27 @@ pub enum MessageType {
     Terminate,
 }
 
+#[derive(Serialize, Debug, PartialEq, PartialOrd)]
+pub enum DisconnectReason {
+    None,
+    ConnectionError,
+    Timeout,
+}
+
+impl fmt::Display for DisconnectReason {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                DisconnectReason::None => "Normal reason",
```

Comment:
nit: would "Client disconnect" or similar be more meaningful to the developer than "Normal reason" as the string here?

Reply:
There are a number of reasons that `None` could be returned for a given disconnect, including locally disconnecting. Still, "Client Disconnect" is a bit more meaningful, so not going to argue.